build_r.R

```diff
@@ -369,6 +369,41 @@ description_contents <- gsub(
 )
 writeLines(description_contents, DESCRIPTION_FILE)
 
+# CMake-based builds can't currently use R's builtin routine registration,
+# so have to update NAMESPACE manually, with a statement like this:
+#
+# useDynLib(lib_lightgbm, LGBM_GetLastError_R, LGBM_DatasetCreateFromFile_R, ...)
```

Comment:
Maybe it will be easier to have this content by default in the file, and **remove** it for CRAN build?

Reply:
hmmm I'm not sure. This looks like more code than having this in the file by default and removing it for CRAN builds, but I thought this R code would be easier to write than the equivalent code with `sed` in `build-cran-package.sh`. Just because it involves replacing / writing literal `.` and literal `)` / `(` and I was intimidated by figuring out how to do the escaping correctly in `sed` haha.

Do you have a strong preference for this change? Otherwise, I'd prefer to leave this as-is for now and maybe make that change as some maintenance work in the future. Just because I'd like to keep pushing forward on #3016 and it doesn't seem like there is anything *wrong* about the current approach.

Reply:
Not really, this was just a proposal to simplify the code. I'm totally fine with the working current approach.

Reply:
ok great, thank you very much!